### DISCUSSION TITLE
課題12についての質問  07/26

### DISCUSSION THREAD

**Discussion by TakedaAyano** [nsc-kiwamu/traning-takeda#51] [Q&A]
①問題３では、勝ち負け判定のエクセルを元に必要のない箇所を削除しました。
その中で、引き分けが２回出ているものはそのままにしてあります。

しかし、ターミナルに出力をしてみると引き分けのところでおかしくなるみたいです。
引き分けなのに「○○さんの勝ちです」と出力されてしまいます。
条件式の[ ]の数字や文字ミスがあるのかと思い、確認してみましたがどこも間違っている気配がしないので、どこが原因でそうなっているのかわかりませんでした。
これもまた、何か必要な式を間違って消しているのでしょうか？

②課題４では、サイコロを振った４人の中の最大値を元に「○○さんの勝ちです」と表示させることができました。
ここから２人勝者が出た場合の「○○さんと○○さんが勝ちです」と表示させたいのですが、この時の条件式は、179行目の条件式「users4[i].matchValue[0] === max」”最大値と同じであれば勝ち”を”２人以上の時”という式を作成しないといけないと考えました。
なので、左右の式が共に正しい時を表す演算子「&&」を使用することで合っていますでしょうか？
２人以上なので、「<」や「>」を使用するのかとも考えましたが、左辺と右辺の大きさを比べてしまうので違うと思いました。

**Comment by nsc-kiwamu** [comment]
回答①
見た感じエクセルが消しすぎているように見えます。
明日の朝会で
・どれを消したのか
・消したところ以外は正しく実装できているか
を確認しましょう

回答②
大分良くなっています。
が、以前指摘した↓の②の内容はクリアできていますか？
https://github.com/nsc-kiwamu/traning-takeda/discussions/50#discussioncomment-3222891

また「○○さんと○○さんが勝ちです」と表示するのは少し手間がかかります。
課題では
```
「Cさんの勝ちです」「Dさんの勝ちです」と出すか「CさんとDさんの勝ちです」と出す
```
としているので、まずは「Cさんの勝ちです」「Dさんの勝ちです」
と出るようにしましょう。
コンソール出力上は
「Cさんの勝ちです」
「Dさんの勝ちです」
という出方で構いません。
